### DISCUSSION TITLE
Executable code environment

### DISCUSSION THREAD

**Discussion by rjbergerud** [athensresearch/athens#1043] [Feature Requests]
This is probably outside the scope of what this application is trying to accomplish.  I have some sort of vague idea of athens meets jupyter notebooks, where you can mix back-linked note-taking and executable cells.

**Comment by jefftangx** [comment]
Right now we are focused on research and notetaking, and users do take notes on math, science, and programming. It makes a lot of sense for us to focus on building an executable code. It's pretty easy to do live JS and CLJS in the browser today. For languages like Python, I would have to look closer, but I think we would need WASM and GraalVM, otherwise we would be re-creating aspects of [repl.it](http://repl.it/).

**Reply by tomisme** [reply]
I'm super excited by the idea of executable code in Athens, I did a tiny demo of using [SCI](https://github.com/borkdude/sci), passing data between blocks and rendering hiccup in [Athen's devcards](https://tomisme.github.io/athens/cards.html#!/athens.devcards.sci_boxes) some time ago. Are the devcards still working btw @tangjeff0? [This page ](https://athensresearch.github.io/athens/cards.html) seems down.

Lots of unanswered questions like, how do I safely expose part of my knowledge graph to a code environment? How might I trust 3rd party code?

I reckon the answers lie in the [Principle of Least Authority](https://en.wikipedia.org/wiki/Principle_of_least_privilege) and things like [Capabilities ](https://en.wikipedia.org/wiki/Capability-based_security).

**Reply by jefftangx** [reply]
@tomisme devcards still compile and open locally, although they are out of date. I'm not sure we publish them to GH pages anymore.
